### ISSUE TITLE
Question: what is the difference between "diagnostics" and "preEmitDiagnostics?"

### ISSUE THREAD

**Issue by anurbol** [dsherret/ts-morph#384]
Couldn't find the answer here: https://dsherret.github.io/ts-simple-ast/setup/diagnostics

In the link there are `project.getDiagnostics` and `project.getPreEmitDiagnostics`. They seem identical. Are there any hidden difference e.g. in performance?

**Comment by anurbol** [comment]
BTW, thanks @dsherret this library saved my life. Default TS compiler API was pain.

**Comment by anurbol** [comment]
Thank you for the response! Cool, I am glad I gave you idea on cleaning the library up. I ended up using 
`getPreEmitDiagnostics()`. If I understand correctly, you will just deprecate the other method and add new methods to `Diagnostic` object item (of Diagnostic[], that is returned from `getPreEmitDiagnostics()`) so there would be no breaking change to API of `getPreEmitDiagnostics()` (I hope it would not be broke). Thank you again! Good luck!
